File: javascript/test/trajpoly5.js
#!/usr/bin/env node
const liba = require("../liba");
function assert(expr) {
    if (!eval(expr)) {
        throw Error(expr);
    }
}
function do_leak_check() {
    for (const arg of arguments) {
        arg.delete();
    }
    if (liba.do_leak_check) {
        liba.do_leak_check();
    }
}
var ctx = new liba.trajpoly5(1, 0, 1, 0, 1, 0, 1);
assert("ctx.c0");
assert("ctx.c1");
assert("ctx.c2");
for (var i = 0; i < 1; i += 0.1) {
    console.log(i, ctx.pos(i), ctx.vel(i), ctx.acc(i));
}
do_leak_check(ctx);

File: lua/test/trajpoly5.lua
#!/usr/bin/env lua

---@diagnostic disable: redefined-local
package.path = arg[0]:sub(0, -arg[0]:match("([^/\\]*)$"):len() - 1) .. "?.lua;" .. package.path
local test = require("test")
local a = require("liba")
test.dir(getmetatable(a.trajpoly5))
local ctx = a.trajpoly5.new(1, 0, 1, 0, 1, 0, 1)
assert(ctx)
test.dir(getmetatable(ctx))
assert(type(a.trajpoly5.pos(ctx, 0.5)) == "number")
assert(type(a.trajpoly5.vel(ctx, 0.5)) == "number")
assert(type(a.trajpoly5.acc(ctx, 0.5)) == "number")
assert(type(ctx.pos(ctx, 0.5)) == "number")
assert(type(ctx.vel(ctx, 0.5)) == "number")
assert(type(ctx.acc(ctx, 0.5)) == "number")
assert(type(ctx:pos(0.5)) == "number")
assert(type(ctx:vel(0.5)) == "number")
assert(type(ctx:acc(0.5)) == "number")
assert(type(ctx.c0) == "table")
assert(type(ctx.c1) == "table")
assert(type(ctx.c2) == "table")
ctx.__name = nil
assert(ctx.__name)
ctx.__index = nil
assert(ctx.__index)
ctx.__newindex = nil
assert(ctx.__newindex)
---@class a.trajpoly5
---@field __name string
---@field __index table
---@field __newindex table
